src/erlcloud_mon.erl

```diff
@@ -216,33 +217,111 @@ put_metric_data(Namespace, MetricName, Value, Unit, Timestamp) ->
+        EndTime     ::datetime() | string(),
+        InstanceId  ::string()
+       ) -> term().
+
+get_metric_statistics(
+  MetricName,
+  StartTime,
+  EndTime,
+  InstanceId) ->
+          get_metric_statistics(
+              "AWS/EC2",
+              MetricName,
+              StartTime,
+              EndTime,
+              60,
+              "",
+              ["Average","Maximum"],
+              [{'InstanceId', InstanceId}]).
+
 %%------------------------------------------------------------------------------
 %% @doc CloudWatch API - GetMetricStatistics
 %% [http://docs.amazonwebservices.com/AmazonCloudWatch/latest/APIReference/index.html?API_GetMetricStatistics.html]
-%% @end
+%%
+%% USAGE:
+%%
+%%  erlcloud_mon:get_metric_statistics(
+%%    "AWS/EC2",
+%%    "CPUUtilization",
+%%    {{2016, 06, 29}, {0, 0, 0}},
+%%    "2016-06-29T00:30:00Z",
+%%    60,
+%%    "Percent",
+%%    ["Average", "Maximum"],
+%%    [{"InstanceType", "t2.micro"}]).
+%%                                     
+%% @end 
 %%------------------------------------------------------------------------------
 -spec get_metric_statistics(
         Namespace   ::string(),
         MetricName  ::string(),
-        StartTime   ::string(),
-        EndTime     ::string(),
+        StartTime   ::datetime() | string(),
+        EndTime     ::datetime() | string(),
         Period      ::pos_integer(),
         Unit        ::string(),
         Statistics  ::[string()],
-        Dimensions  ::[string()]
+        Dimensions  ::[{string(), string()}]
                       ) -> term().
 
 get_metric_statistics(
-  _Namespace,
-  _MetricName,
-  _StartTime,
-  _EndTime,
-  _Period,
-  _Unit,
-  _Statistics,
-  _Dimensions
+  Namespace,
+  MetricName,
+  StartTime,
+  EndTime,
+  Period,
+  Unit,
+  Statistics,
+  Dimensions
  ) ->
-    todo.
+    Config = default_config(),
+    Params =
+          lists:flatten([
+           {"Namespace",  Namespace},
+           {"MetricName", MetricName},
+           {"StartTime",  format_timestamp(StartTime)},
+           {"EndTime",    format_timestamp(EndTime)},
+           {"Period",     Period},
+           [{"Unit",      Unit} || Unit=/=undefined, Unit=/=""]
+          ])
+          ++
+          lists:flatten(
+            [begin
+                 Value = lists:nth(N, Statistics),
+                 [{?FMT("Statistics.member.~b", [N]), Value}]
+             end
+             || N<-lists:seq(1, length(Statistics))]
+           )
+          ++
+          lists:flatten(
+            [begin
+                 {Name, Value} = lists:nth(N, Dimensions),
+                 [{?FMT("Dimensions.member.~b.Name", [N]), Name},
+                  {?FMT("Dimensions.member.~b.Value", [N]), Value}]
+             end
+             || N<-lists:seq(1, length(Dimensions))]
+           ),
+    Doc = mon_query(Config, "GetMetricStatistics", Params),
+    Members = xmerl_xpath:string("/GetMetricStatisticsResponse/GetMetricStatisticsResult/Datapoints/member", Doc),
```

Comment:
@motobob it looks like Label is just what we send as MetricName, and we must send exactly MetricName. Including that will put the useful datapoint results one level deeper into the returned object, since there has to be the top level Label string and then the DataPoint list.

Anyway if you think it's the right move, I did it, so I'll push it if you say the word. The results look like this:
`{{'label', 'CPUUtilization'}, {'datapoints', [[{:timestamp, {{2016, 6, 29}, {0, 3, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, {'maximum', '0.0'}], [{:timestamp, {{2016, 6, 29}, {0, 1, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, {'maximum', '0.0'}], [{:timestamp, {{2016, 6, 29}, {0, 0, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, {'maximum', '0.0'}], [{:timestamp, {{2016, 6, 29}, {0, 11, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, {'maximum', '0.0'}], [{:timestamp, {{2016, 6, 29}, {0, 6, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, {'maximum', '0.0'}], [{:timestamp, {{2016, 6, 29}, {0, 16, 0}}}, {:unit, 'Percent'}, {'average', '0.0'}, ...`

Also, should each datapoint have a list around it like this? It looks redundant, but it also looks like this is what dict:from_list wants.